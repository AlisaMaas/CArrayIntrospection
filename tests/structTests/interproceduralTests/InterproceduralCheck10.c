#include "structHeader.h"
/**
 * This tests that function declarations and functions without arguments
 * don't cause issues.
 **/
void print(char *string);
void nonsense(){}
int find(struct foo *f) {
	print(f->nonString);
	for (int i = 0;; i++) {
		if (f->string[i] == '\0') {
			print("About to break from loop");
			break;
		}
	}
	nonsense();
	return 1;
}
int foo(struct foo *f) {
	print(f->nonString);
	for (int i = 0;; i++) {
		if (f->nonString[i] != '\0') {
			print("About to break from loop");
			break;
		}
	}
	return find(f);
}
