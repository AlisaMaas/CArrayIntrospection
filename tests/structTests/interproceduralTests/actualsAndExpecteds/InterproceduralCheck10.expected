struct struct.foo[0] should be annotated NULL_TERMINATED (Sentinel-terminated by 0) because found a non optional sentinel check for string in find.
